lib/JitEngineDeviceHIP.cpp

```diff
@@ -132,6 +133,79 @@ void JitEngineDeviceHIP::setLaunchBoundsForKernel(Module &M, Function &F,
              << WavesPerEU << "\n");
 }
 
+void JitEngineDeviceHIP::setKernelDims(Module &M, dim3 &GridDim,
```

Comment:
There is a lot of code duplication between the HIP and CUDA cases. Move `setKernelDims` to `JitEngineDevice.hpp`. Create CRTP specialized getters for dimension functions that return the instrinsic/function name.